extension/llm/export/builder.py

```diff
@@ -167,6 +176,59 @@ def capture_pre_autograd_graph(self) -> "LLMEdgeManager":
             )
         return self
 
+
+    def pt2e_calibrate(
+        self,
+        prepared_module,
+        calibration_tasks,
+        calibration_limit,
+        calibration_seq_length,
+        tokenizer_path,
+    ):
+        logging.info("Run calibration...")
+        try:
+            from executorch.examples.models.llama2.evaluate import EagerEvalWrapper, evaluate_model
+        except ImportError:
+            raise ImportError(
+                "Please install the llm eval dependency via examples/models/llama2/install_requirements.sh"
+            )
+
+        tokenizer = get_tokenizer(tokenizer_path)
+
+        def calibrate_template(module: torch.fx.GraphModule, tokenizer, string: str = "Once upon a time", max_len: int = 128):
+                # TODO: change criteria & support batch inputs if necessary
+                pos = torch.tensor(0, dtype=torch.int64)
+                token_list = [tokenizer.bos_id] + tokenizer.encode(string, bos=True, eos=False)
+
+                with torch.no_grad():
+                    while token_list[-1] != tokenizer.eos_id and pos < max_len:
+                        logits = module(
+                            torch.full((1, 1), token_list[pos]),
+                            torch.tensor((pos, )),
+                        )
+                        pos += 1
+                        if pos >= len(token_list):
+                            token_list.append(torch.argmax(logits[:], dim=-1).item())
+
+        calibrate_template(prepared_module, tokenizer, string="Once upon a time", max_len=calibration_seq_length)
```

Comment:
Maybe we could add an argument to calibrate user specific prompts or data?

Reply:
That's a good idea!